solidity/dashboard/src/sagas/messages.js

```diff
@@ -5,13 +5,34 @@ import {
   REMOVE_MESSAGE,
   ADD_MESSAGE,
 } from "../actions/messages"
+import { messageType } from "../components/Message"
 
 function* addMessage(action) {
   yield put({ type: ADD_MESSAGE, payload: action.payload })
 }
 
 function* removeMessage(action) {
-  yield put({ type: REMOVE_MESSAGE, payload: action.payload })
+  console.log("action", action)
+  yield put({ type: REMOVE_MESSAGE, payload: action.payload.id })
+  if (action.messageType) {
+    // switch for future notification data clearance for other message types
+    switch (action.payload.messageType) {
+      case messageType.LIQUIDITY_REWARDS_EARNED:
+        yield put({
+          type:
+            "notifications_data/liquidityRewardNotification/pairs_displayed_updated",
+          payload: [],
+        })
+        break
+      default:
+        break
+    }
+  }
+  yield put({
```

Comment:
Tbh I don't understand why we are dispatching `notifications_data/liquidityRewardNotification/pairs_displayed_updated` action . Could you please explain?

Reply:
Sure. So in this PR we only have one notification of liquidity rewards which displays all the pairs that the user get rewards for (see screenshot that i uploaded on this PR). These pairs are stored in redux store, so to update the pairs we are dispatching the action that you mentioned. This case we don't need to display several notifications for all the pairs, but only one notification, and update the pairs in redux store.

Here we dispatch the `notifications_data/liquidityRewardNotification/pairs_displayed_updated` action and put an empty array in the payload, because we want to clear the data when user closes the message. If he closes it, then the next time we will check if he got any rewards for all the pairs and update the state. If he doesn't close it though, then we keep the pairs already displayed in the notification (the store is not empty) and check if he got any rewards for the pairs that are not displayed.

I tried to write it as simple as possible. Hope it worked out 🙈

Reply:
I understand what this action is responsible for but I don't understand why we dispatch this action even though we handle it in the `switch` statement.

Reply:
Oh, i see! 😮 You're right! We shouldn't dispatch the action again after the `switch` statement.